### PR TITLE
[DOCS] Reformats cat segments API

### PR DESCRIPTION
**Pull request by jrodewig**
This PR updates the cat segments API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Relates to elastic/docs#937 and #45196

### Preview
http://elasticsearch_45397.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/cat-segments.html

### CODE REVIEW

File: docs/reference/cat/segments.asciidoc [hunk 1]
```diff
@@ -1,73 +1,123 @@
...
-                The name of the segment is derived from this generation number.
+`docs.count`::
+(Default) Number of non-deleted documents in the segment, such as `25`. This
+number is based on Lucene documents and may include documents from
+<<nested,nested>> fields.
+                
+`docs.deleted`::
+(Default) Number of deleted documents in the segment, such as `0`. This number
+is based on Lucene documents. {es} reclaims the disk space of Lucene documents a
+segment is merged.
```
Comment (debadair) [inline docs/reference/cat/segments.asciidoc]: I think this last sentence is missing a "when"? Not sure I'd repeat "Lucene" -- I think the more important point is that the disk space from _deleted_ documents is reclaimed when segments are merged.

**Review by debadair — CHANGES_REQUESTED** [review]
Left one minor comment.

File: docs/reference/cat/segments.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/cat/segments.asciidoc]: Thanks @debadair. Fixed with 96a1d9c858e.

**Review by debadair — APPROVED** [review]
LGTM

**Pull request merged**